src/com/pentalog/pentastagiu/homework2/shop/Product.java

```diff
@@ -0,0 +1,42 @@
+package com.pentalog.pentastagiu.homework2.shop;
+
+public abstract class Product {
+    private long id;
+    private String name;
+    private int price;
+
+    public Product(long id, String name, int price) {
+        this.id = id;
+        this.name = name;
+        this.price = price;
```

Comment:
in constructori si setteri/metode(publce) trebuie sa te asiguri ca nu lasi sa treaca valori invalide dpdv al businessului, de ex pret si cantitate, nu ar trebui sa acceptam valori negative si e mult mai bine sa aruncam o exceptie din constructor daca gasim date invalide decat sa lasam sa se creeze o instanta si sa crape mai incolo in cod